arkouda/groupbyclass.py

```diff
@@ -2092,14 +2101,21 @@ def broadcast(
     if size < 1:
         raise ValueError("result size must be greater than zero")
     cmd = "broadcast"
-    repMsg = generic_msg(
-        cmd=cmd,
-        args={
-            "permName": pname,
-            "segName": segments.name,
-            "valName": values.name,
-            "permute": permute,
-            "size": size,
-        },
+    repMsg = cast(
+        str,
+        generic_msg(
+            cmd=cmd,
+            args={
+                "permName": pname,
+                "segName": segments.name,
+                "valName": values.name,
+                "objType": values.objType,
+                "permute": permute,
+                "size": size,
+            },
+        ),
     )
-    return create_pdarray(repMsg)
+    if values.objType == Strings.objType:
+        return Strings.from_return_msg(repMsg)
+    else:
+        return create_pdarray(repMsg)
```

Comment:
Based on this it looks like the `values` parameter can be `pdarray` or `Strings`. However, the function definition of `values: pdarray` will not allow this.